spec/requests/v1/ont/runs_spec.rb

```diff
@@ -60,6 +60,348 @@ def json
         expect(fc2.pool.id).to eq(attr2[:ont_pool_id])
       end
     end
+
+    context 'on failure' do
```

Comment:
Ah. Because you have checked each failure in the model you only need to test that the response is not successful and it has some errors. So you only need one failure test. Also as I have added message publishing which I have just merged into develop you will also need to check that a message is not published.

Reply:
I have removed the redundant tests, merged develop into the branch and added Messages-not-published tests to error cases.